packages/nimble-components/src/table/types.ts

```diff
@@ -40,6 +40,18 @@ export interface TableRecord {
     [key: TableFieldName]: TableFieldValue;
 }
 
+/**
+ * InternalTableRecord describes a hierarchical data structure that is used for
+ * the internal representation of the data, and allows us to represent data with
+ * parent-child relationships within Tanstack.
+ */
+export interface InternalTableRecord<
```

Comment:
I'm guessing we can save index on InternalTableRecords in the future right? like `userIndex` for the `userRecord`?

Reply:
Based on the other convo `clientRecord` and `clientRecordIndex` :)

Reply:
I don't think we can do that through this mechanism unfortunately. This interface needs to mirror what the 3rd party `arrayToTree` method produces. I handled tracking the index in the prototype by keeping a map that associated row Id with a set of metadata, one piece of which was the original index in the array.

Assuming there is more to discuss on this topic, my understanding is that this doesn't need to hold up this PR.